doc/src/manual/multi-threading.md

```diff
@@ -72,7 +72,50 @@ julia> Threads.threadid()
     three processes have 2 threads enabled. For more fine grained control over worker
     threads use [`addprocs`](@ref) and pass `-t`/`--threads` as `exeflags`.
 
-## Data-race freedom
+## [Threadpools](@id man-threadpools)
+
+When a program's threads are busy with many tasks to run, tasks may experience
+delays which may negatively affect the responsiveness and interactivity of the
+program. To address this, you can specify that a task is interactive when you
+[`Threads.@spawn`](@ref) it. Julia may be started with one or more threads
+reserved to run interactive tasks:
+
+```bash
+$ julia --threads 3,1
+```
```

Comment:
I wonder if we should implement and advertise `julia --threads 3,auto` and allocate an unspecified number of worker threads for the second pool. It's not clear if 1 is the best number for this kind of use case and we don't want to change people's habits after figuring out the best strategies. (Of course, I think it's OK to let users specify the exact number if they want to tune their application.)

For example, [Python uses `min(32, os.cpu_count() + 4)`](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ThreadPoolExecutor) for a similar setting and I think it makes sense to use a similar strategy when `auto` is specified. This worker pool should be mostly silent and the task should be quickly executed once it's enqueued. I think it's reasonable to allocate a relatively large amount of worker threads to reduce the latencies even if the scheduled tasks are not very cooperative.

Reply:
Thread wakeup latency will hurt the 'low latency' contract; I'm not sure there's going to be any benefit from having many threads in the interactive threadpool. Also, I'm dubious about the concept of `auto`, e.g. #43692. There are many situations where adding threads hurts performance -- automatically having multiple threads is not always a good idea.

Reply:
So `auto` would mean `Sys.CPU_THREADS` for the default threadpool and `1` for the secondary threadpool? Then, `julia --threads=auto,auto` would be `Sys.CPU_THREADS - 1` and `1`? I guess we could do that. Currently, this doesn't support `auto` if a secondary threadpool is specified, i.e. `julia --threads=auto,1` wouldn't work and that seems useful... but perhaps `julia --threads=auto` should do that?

Reply:
I agree with @tkf , making a reasonable automatic choice prevents the very socially-costly retraining when changing defaults.

Reply:
I spoke with Taka on Slack, and I also agree. The remaining question is: should we support `julia --threads=auto|N[,auto|M]` where `auto,auto` means `Sys.CPU_THREADS-1,1` or should it be `julia --threads=auto` which would mean the same thing?

If people are using `julia --threads=auto` in performance-sensitive runs, then taking away 1 thread to use for the interactive threadpool might cause dismay. My guess would be that for performance-sensitive runs, people are more likely to be specific about how many threads to use, but I'm not sure...

Reply:
I think it's reasonable to expect users to opt-in to having an extra thread for now (yes, an *extra* thread, not trading one for another). I think we also should keep the default threadpool at `Sys.CPU_THREADS`, and just have 1 additional thread for interactive uses, if opted-into.

My understanding is that I/O is typically looked favorably upon by OS schedulers, and will cause high-I/O threads to get more compute time versus a thread which rarely does I/O. This should result in the interactive thread getting execution time when it needs it, but as long as the interactive work is truly low-latency and "sparse", it shouldn't significantly impact the default threadpool's performance. So adding an extra thread shouldn't significantly bog down performance sensistive, high-latency operations.

Reply:
So `julia --threads=auto,auto` would oversubscribe by having `Sys.CPU_THREADS,1`? I'm pretty sure that OS thread context switch overhead will kill any gains from having the interactive thread in the first place.

Reply:
Okay, done. Now it is: `-t, --threads {auto|N[,auto|M]}` where the second `auto` means 1.